godtools/App/Share/Data/ResourceBannerImageRepository/ResourceBannerImageRepository.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  ResourceBannerImageRepository.swift
+//  godtools
+//
+//  Created by Levi Eggert on 6/28/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import UIKit
+import SwiftUI
+
+class ResourceBannerImageRepository {
+    
+    private let attachmentsFileCache: AttachmentsFileCache
+    private let bundleBannerImages: BundleResourceBannerImages
+    
+    init(attachmentsFileCache: AttachmentsFileCache, bundleBannerImages: BundleResourceBannerImages) {
+        
+        self.attachmentsFileCache = attachmentsFileCache
+        self.bundleBannerImages = bundleBannerImages
+    }
+    
+    func getBannerUIImage(resourceId: String, bannerId: String) -> UIImage? {
+        
+        if let cachedImage = attachmentsFileCache.getAttachmentBanner(attachmentId: bannerId) {
+            
+            return cachedImage
+        }
+        else if let bundleImage = bundleBannerImages.getBannerImage(resourceId: resourceId) {
+            
+            return bundleImage
+        }
+        
+        return nil
+    }
+    
+    func getBannerImage(resourceId: String, bannerId: String) -> Image? {
```

Comment:
Nice! 🎉 
Do we need a followup ticket to go back and refactor other SwiftUI views to use this instead of DeviceAttachmentsBanner?

Reply:
Yep that will be this ticket here -> https://jira.cru.org/browse/GT-1653?atlLinkOrigin=c2xhY2staW50ZWdyYXRpb258aXNzdWU%3D